lib/wasi/src/state/mod.rs

```diff
@@ -128,10 +128,18 @@ pub struct Fd {
 }
 
 impl Fd {
```

Comment:
Are there any invariants across the whole group? Like, all file descriptors must be open for at least reading or writing?